src/launch/service.c

```diff
@@ -260,7 +254,13 @@ static int service_watch_jobs_handler(sd_bus_message *message, void *userdata, s
                  * Our job failed. Forward this information to the broker so it
                  * can fail pending activations.
                  */
-                r = service_reset_activation(service);
+                _c_cleanup_(c_freep) char *error_message = NULL;
+
+                r = asprintf(&error_message, "job result: %s", result);
+                if (r < 0)
+                        return error_origin(-errno);
+
+                r = service_reset_activation(service, error_message);
```

Comment:
I think we should define this as something like `org.bus1.Error.<something matching the error we received>`.  If we also want to include a received human-readable error message (in case that's what systemd gives us), I think that's fine, but it should be secondary, I think.

Reply:
given these are strings, then we'll need to hard-code the currently known one and convert to the enum - is that ok?

Reply:
I would be fine with `org.bus1.DBus.Name.Error.StartupFailure` regardless of the exact type. But yeah, if we want it fine-grained, we need to add a strcmp() for each known result and convert it into a proper error-name.

Reply:
Maybe I misunderstood, so take this with a grain of salt, but I would be against matching on free form text, but in favor of matching on reverse domain name error names if we have them to make it more finegrained.